StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/UpdatePaymentMethodViewModel.swift

```diff
@@ -82,4 +79,37 @@ class UpdatePaymentMethodViewModel {
         self.canSetAsDefaultPM = allowsSetAsDefaultPM && !isDefault
         self.isDefault = isDefault
     }
+
+    func updateParams(paymentMethodElement: PaymentMethodElement) -> UpdatePaymentMethodOptions?{
+        let confirmParams = IntentConfirmParams(type: PaymentSheet.PaymentMethodType.stripe(.card))
+
+        if let params = paymentMethodElement.updateParams(params: confirmParams),
+           let cardParams = params.paymentMethodParams.card,
+           let originalPaymentMethodCard = paymentMethod.card,
+           hasChangedFields(original: originalPaymentMethodCard, updated: cardParams) {
+            return .card(paymentMethodCardParams: cardParams)
+        }
+        return nil
+    }
+    func hasChangedFields(original: STPPaymentMethodCard, updated: STPPaymentMethodCardParams) -> Bool {
```

Comment:
This method name implies it merely calculates something (whether or not the fields have changed), which is a great single purpose method. I'm not sure it should also send a 'card brand selected' analytic - that seems like an unrelated responsibility. Why are these two distinct things combined?

I'd expect the function that handles 'card brand was selected' (something that handles the dropdown value being changed) to send the 'card brand selected' analytic, is that possible?

Reply:
Agree. I initially had this in the factory, but since that was changed to a static function, I needed some way to keep track of what brand we've logged already (`lastCardBrandLogSelectedEventSent`).

In haste, I moved this functionality over, but you're totally right. So, to remedy this, I've added a function to the viewModel to take care of sending this metric.